packages/repository/src/relations/has-one/has-one.repository.ts

```diff
@@ -39,6 +40,22 @@ export interface HasOneRepository<Target extends Entity> {
     filter?: Pick<Filter<Target>, Exclude<keyof Filter<Target>, 'where'>>,
     options?: Options,
   ): Promise<Target>;
+
+  /**
+   * Delete multiple target model instances
+   * @param where Instances within the where scope are deleted
+   * @param options
+   * @returns A promise which resolves the deleted target model instances
+   */
+  delete(where?: Where<Target>, options?: Options): Promise<Count>;
```

Comment:
Since there is only one (at most one) target model instance to be deleted, I feel the `where` argument does not make sense. Let's remove it.

```suggestion
  delete(options?: Options): Promise<Count>;
```

Reply:
Actually, the hasOne relation allows to create two targets for a same source. This is what I wrote in my previous comments. In my opinion, delete operation shall remove all targets associated to a same source. When we will fix this issue to avoid to create two targets for a same source (new PR) then we can remove the Where as you recommend.
What do you think, can we keep the where till the fix?